kuzzle-sdk-java/src/main/java/io/kuzzle/sdk/Kuzzle.java

```diff
@@ -0,0 +1,284 @@
+package io.kuzzle.sdk;
+
+import io.kuzzle.sdk.CoreClasses.Json.JsonSerializer;
+import io.kuzzle.sdk.CoreClasses.Maps.KuzzleMap;
+import io.kuzzle.sdk.CoreClasses.Task;
+import io.kuzzle.sdk.Events.EventListener;
+import io.kuzzle.sdk.Exceptions.*;
+import io.kuzzle.sdk.Options.KuzzleOptions;
+import io.kuzzle.sdk.Protocol.AbstractProtocol;
+import io.kuzzle.sdk.Protocol.ProtocolState;
+
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+import io.kuzzle.sdk.CoreClasses.Responses.*;
+
+import static io.kuzzle.sdk.Helpers.Default.notNull;
+
+public class Kuzzle {
+
+    protected EventListener tokenExpiredEvent;
+    protected EventListener<Response> unhandledResponseEvent;
+
+    protected final AbstractProtocol networkProtocol;
+
+    public final String version;
```

Comment:
The SDK should have a `sdkName` property to send alongside the `instanceId` in the volatile headers so Kuzzle is able to reject the request if it come from an incompatible SDK version.
Format: `java@3.0.0`

Reply:
Yes it can be discussed but since it's already implemented in Kuzzle we should talk about this elsewhere (CF: https://github.com/kuzzleio/kuzzle/pull/1495).
Also I think it will be difficult to change this before Kuzzle v3 :/